src/AnalysisManager.cpp

```diff
@@ -27,320 +48,159 @@ AnalysisManager::~AnalysisManager()
-    generator_initial_particle_pz_.clear();
-    generator_initial_particle_energy_.clear();
-    generator_initial_particle_pdg_code_.clear();
-    generator_initial_particle_mass_.clear();
-    generator_initial_particle_charge_.clear();
-
-    generator_final_number_particles_ = 0;
-    generator_final_particle_x_.clear();
-    generator_final_particle_y_.clear();
-    generator_final_particle_z_.clear();
-    generator_final_particle_t_.clear();
-    generator_final_particle_px_.clear();
-    generator_final_particle_py_.clear();
-    generator_final_particle_pz_.clear();
-    generator_final_particle_energy_.clear();
-    generator_final_particle_pdg_code_.clear();
-    generator_final_particle_mass_.clear();
-    generator_final_particle_charge_.clear();
-
-    number_hits_ = 0;
-    energy_deposit_ = 0;
-
-    particle_track_id_.clear();
-    particle_parent_track_id_.clear();
-    particle_pdg_code_.clear();
-    particle_mass_.clear();
-    particle_charge_.clear();
-    particle_process_key_.clear();
-    particle_total_occupancy_.clear();
-
-    particle_number_daughters_.clear();
-    particle_daughter_track_ids_.clear();
+  if (!G4Threading::IsMasterThread()) {return;}  // only run Save() with the master thread
 
-    particle_initial_x_.clear();
-    particle_initial_y_.clear();
-    particle_initial_z_.clear();
-    particle_initial_t_.clear();
-
-    particle_initial_px_.clear();
-    particle_initial_py_.clear();
-    particle_initial_pz_.clear();
-    particle_initial_energy_.clear();
-
-    hit_track_id_.clear();
-    hit_start_x_.clear();
-    hit_start_y_.clear();
-    hit_start_z_.clear();
-    hit_start_t_.clear();
-    hit_end_x_.clear();
-    hit_end_y_.clear();
-    hit_end_z_.clear();
-    hit_end_t_.clear();
-    hit_energy_deposit_.clear();
-    hit_length_.clear();
-    hit_process_key_.clear();
-}
-
-//-----------------------------------------------------------------------------
-void AnalysisManager::EventFill()
-{
-    // fill TTree objects per event
-    event_tree_->Fill();
-}
+  //G4AutoLock saveLock(&saveMutex);
 
-//-----------------------------------------------------------------------------
-void AnalysisManager::SetRun(int const value)
-{
-    run_ = value;
+  // write TTree objects to file and close file
+  tfile_->cd();
+  metadata_->Write();
+  event_tree_->Write();
+  tfile_->Close();
 }
 
 //-----------------------------------------------------------------------------
-void AnalysisManager::SetEvent(int const value)
+void AnalysisManager::EventFill(const AnalysisData& rhs)
 {
-    event_ = value;
+  G4AutoLock fillLock(&fillMutex);
+  // fill TTree objects per event
+  event = rhs;
+  event_tree_->Fill();
 }
 
 //-----------------------------------------------------------------------------
-void AnalysisManager::FillMetadata(double const & detector_length_x,
+void AnalysisManager::FillMetadata(bool const & detectorConfiguration, double const & detector_length_x,
```

Comment:
The detectorConfiguration argument is no longer needed if we use the ConfigManager.